docs/src/content/docs/book/receive.mdx

```diff
@@ -8,7 +8,11 @@ prev:
 
 import { Badge } from '@astrojs/starlight/components';
 
-TON is a distributed blockchain, which means that communication between contracts is performed by sending and receiving messages. The most common type of message is the _internal message_ — a message sent from one contract (or a wallet) to another.
+TON is a distributed blockchain, which means that communication between smart contracts is performed by sending and receiving messages. The most common type of message is the _internal message_ — a message sent from one contract (or a wallet) to another.
+
+A single received message equates to a single transaction. That is, no other message can influence the flow of the current transaction initiated by the incoming message. On top of that, due to the [split into different phases](/book/send#outbound-message-processing), outgoing messages are not sent until the end of computations within the current transaction. Thus, reentrancy of messages is not possible.
```

Comment:
just mention explicitly that sending messages does not invoke any external code immediately but instead queues it